### ISSUE TITLE
Crash when running LCIO event generator with crossing angle

### ISSUE THREAD

**Issue by chekanov** [slaclab/slic#16]
Hi,

The version of slic from Ap.17 (master) has failed on single pions with this error:
``
<<< Reference Physics List QGSP_BERT is built

Could not find particle.tbl!
PrimaryGeneratorAction is already registered!
RunAction is already registered!
EventAction is already registered!
SteppingAction is already registered!

********************
* Module Registry  *
********************
CommandLineProcessor
CommandQueue
EventAction
EventSourceManager
FieldManager
HepPDTManager
LcioEventSource
LcioFileNamer
LcioHitsCollectionBuilder
LcioManager
MCParticleManager
ParticleGunEventSource
PhysicsListManager
PrimaryGeneratorAction
RunAction
RunManager
SlicApplication
SlicMain
SteppingAction
TrackManager
Set Lorentz transformation angle to 7 mrad
G4VUserPhysicsList::PreparePhysicsTable  : No Process Manager for B+
B+ should be created in your PhysicsList

-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : Run0273
      issued by : G4VUserPhysicsList::PreparePhysicsTable
No process manager
*** Fatal Exception *** core dump ***
-------- EEEE -------- G4Exception-END --------- EEEE -------

*** G4Exception: Aborting execution ***
./A_RUN: line 142: 4173985 Aborted                 ${SLIC_DIR}/bin/slic -x -i ${FTRUTH} -g ./${GEOM}/${GEOM}.lcdd -r $Nevents -m ./${GEOM}/config/defaultILCCrossingAngle.mac -o ${OUTF}.slcio
HepSim: SLIC took 5 seconds
``

**Comment by JeremyMcCormick** [comment]
I think you need to provide the path to the table of extra particles provided by HepPDT which has the B+ definition in it:

```
slic -P data/particle.tbl [...]
```

This is probably something that could be handled in a more automatic way but for now it is manually specified.

**Comment by chekanov** [comment]
Hi, Jeremy

I've used this suggestion:

  -P  ${SLIC_DIR}/data/particle.tbl


it still breaks on that file I've sent you:

  Set Lorentz transformation angle to 7 mrad
G4VUserPhysicsList::PreparePhysicsTable  : No Process Manager for B+
B+ should be created in your PhysicsList

-------- EEEE ------- G4Exception-START -------- EEEE -------
*** G4Exception : Run0273
       issued by : G4VUserPhysicsList::PreparePhysicsTable
No process manager
*** Fatal Exception *** core dump ***
-------- EEEE -------- G4Exception-END --------- EEEE -------


*** G4Exception: Aborting execution ***
./A_RUN: line 142: 4194018 Aborted                 ${SLIC_DIR}/bin/slic 
-P ${SLIC_DIR}/data/particle.tbl -x -i ${FTRUTH} -g 
./${GEOM}/${GEOM}.lcdd -r $Nevents -m 
./${GEOM}/config/defaultILCCrossingAngle.mac -o ${OUTF}.slcio
HepSim: SLIC took 5 seconds


       Sergei Chekanov           | http://www.hep.anl.gov/chekanov/
       ===============           | Email: chekanov@anl.gov
      HEP Division, ANL          | Tel  : +01-630-252 6541
9700 S.Cass,Lemont, IL60439,USA | Fax  : +01-630-252-5782

**Comment by JeremyMcCormick** [comment]
Okay, I will have to check this more carefully.  It seems that Geant4 10.3 broke a number of things in the initialization that used to work in 10.2 and earlier.  So I'll need to go over the run manager code to make sure that the initialization is all happening properly.  In the meantime it is possible this would all work better with a previous Geant4 version (10.1.x or 10.2.x).

**Comment by chekanov** [comment]
Hi, Jeremy

Take  a look at Whit's SLIC version I've sent you by email. Simply run 
"meld" on "src" of these versions. His version works OK

best, Sergei

       Sergei Chekanov           | http://www.hep.anl.gov/chekanov/
       ===============           | Email: chekanov@anl.gov
      HEP Division, ANL          | Tel  : +01-630-252 6541
9700 S.Cass,Lemont, IL60439,USA | Fax  : +01-630-252-5782

**Comment by JeremyMcCormick** [comment]
I insist we do things the "github way" from now on with this project.  This means if you have a patch, then you should fork the slic repo into your own area, commit/push your changes, and then make a PR on your fork for me to review.

This is a far better way of working, as it allows me to easily see what has changed by looking at the PR.  Otherwise, if I get a source tarball by email, I don't know which files changed, what version of the code you are working from, how to integrate those changes exactly or what command to execute to merge them in, etc.

If you have any questions about this process, please let me know.  (It the standard way of working for github projects.)

**Comment by JeremyMcCormick** [comment]
I need to have a public link to an input file that shows this problem.  Please put the slic command line you are executing here too.

Thanks.

**Comment by JeremyMcCormick** [comment]
Is this the file you are using where you see this problem?

[pgun_pi10gev_001_truth.slcio.zip](https://github.com/slaclab/slic/files/940318/pgun_pi10gev_001_truth.slcio.zip)

**Comment by chekanov** [comment]
Hi, Jeremy

  Yes, this is LCIO file with single pions

best, Sergei

       Sergei Chekanov           | http://www.hep.anl.gov/chekanov/
       ===============           | Email: chekanov@anl.gov
      HEP Division, ANL          | Tel  : +01-630-252 6541
9700 S.Cass,Lemont, IL60439,USA | Fax  : +01-630-252-5782

**Comment by chekanov** [comment]
Here is my command to check this version:
FTRUTH=pgun_pi10gev_001_truth.slcio
${SLIC_DIR}/slic/bin/slic -P  ${SLIC_DIR}/data/particle.tbl -x -i pgun_pi10gev_001_truth.slcio
 -g  ./${GEOM}/${GEOM}.lcdd -r $Nevents \
 -m ./${GEOM}/config/defaultILCCrossingAngle.mac -o ${OUTF}.slcio

**Comment by JeremyMcCormick** [comment]
This seems to work fine for me:

```
slic -g ./sidloi3.lcdd -P particle.tbl -i pgun_pi10gev_001_truth.slcio -r 2
```

Are you sure you using an up-to-date copy of slic master?

**Comment by JeremyMcCormick** [comment]
Looks like maybe there is something wrong with the first event (either issue with slic or the first event is empty):

```
>>>> BeginEvent <0>
TrackManager processing 0 TrackSummary objects.
Saved 0 MCParticles.
SiTrackerEndcapHits has 0 hits
EcalEndcapHits has 0 hits
HcalBarrelHits has 0 hits
MuonBarrelHits has 0 hits
SiVertexBarrelHits has 0 hits
HcalEndcapHits has 0 hits
LumiCalHits has 0 hits
SiTrackerBarrelHits has 0 hits
MuonEndcapHits has 0 hits
SiVertexEndcapHits has 0 hits
SiTrackerForwardHits has 0 hits
EcalBarrelHits has 0 hits
BeamCalHits has 0 hits
```

**Comment by chekanov** [comment]
Hi, Jeremy

I usually put empty event "0", since slic skips  it
(for older versions)

best, Sergei



       Sergei Chekanov           | http://www.hep.anl.gov/chekanov/
       ===============           | Email: chekanov@anl.gov
      HEP Division, ANL          | Tel  : +01-630-252 6541
9700 S.Cass,Lemont, IL60439,USA | Fax  : +01-630-252-5782

**Comment by JeremyMcCormick** [comment]
I think it is reading the first event fine (I didn't check the output closely but saw it does make hits for events after the first one).

Are you able to get the slic master working?  Perhaps this issue has gone away now...

**Comment by chekanov** [comment]
Hi, Jeremy

  Thanks for fixing. This is interesting. This command works:

  slic/bin/slic -P particle.tbl -x -i pgun_pi10gev_001_truth.slcio  -g 
./sifcch7/sifcch7.lcdd -r 3


But my usual method is to run it as:

  slic/bin/slic -P  particle.tbl -x -i 
output/pgun_pi10gev_001_truth.slcio  -g  ./sifcch7/sifcch7.lcdd -r 3 -m 
./sifcch7/config/defaultILCCrossingAngle.mac -o 
output/pgun_pi10gev_001.slcio


So, it fails when I add the crossing angle, i.e. the file 
"defaultILCCrossingAngle.mac" with:

/run/initialize
/generator/setLorentzTransformationAngle 7 mrad
/run/beamOn 1

can this be fixed?

best, Sergei


       Sergei Chekanov           | http://www.hep.anl.gov/chekanov/
       ===============           | Email: chekanov@anl.gov
      HEP Division, ANL          | Tel  : +01-630-252 6541
9700 S.Cass,Lemont, IL60439,USA | Fax  : +01-630-252-5782

**Comment by JeremyMcCormick** [comment]
This works fine for me:

```
slic -g ./sidloi3.lcdd -i ./pgun_pi10gev_001_truth.slcio -m xangle.mac -P ./particle.tbl -r 5
```

Where `xangle.mac` contains:

```
/generator/setLorentzTransformationAngle 7 mrad
```

It looks like the issue is that your macro contains `/run/initialize` which it should not because SLIC calls this automatically.

**Comment by chekanov** [comment]
Hi, Jeremy

  Ok, thanks. I confirm - if I remove "initialize" for the crossing 
angle, it works.

The lcsim version, however, is unlikely the problem. I've tried to open 
LCIO file generated with this version of SLIC using Jas with old version 
of LCSIM, and I saw a crash again.

Maybe we can focus now on fixing LCSIM

best, Sergei

         Sergei Chekanov            | https://www.hep.anl.gov/chekanov/
         ===============            | Email: chekanov@anl.gov
        HEP Division, ANL           | Tel  : +01-630-252 6541
  9700 S.Cass,Argonne,IL 60439,USA  | Fax  : +01-630-252-5782
